### ISSUE TITLE
no-relative-imports false positives

### ISSUE THREAD

**Issue by westarne** [qDanik/eslint-plugin-path#24]
I've upgraded to 2.1.0 from 2.0.3. From that moment on I've received false positive errors for nearly or maybe every import like the following:
```
  2:1  error  Relative import path '@/src/auth/Auth.model' should be replaced with '@/src/auth/Auth.model'  path/no-relative-imports
```

As you can see, both paths are identical. Still it breaks. Running eslint with "fix" also detects circular fixes.

Downgrading back to `2.0.3` fixes the issue. I'm using eslint 10 (`10.2.0`) and eslint `9.39.1` in two different projects, both same issue.
Same error with the plugin version `3.0.0-beta.0`.

**Comment by kAIPraxisBot** [comment]
Reproduced and fixed in #25.

The rule itself did not change between `2.0.3` and `2.1.0` — the resolver did. `2.0.3` resolved a specifier by joining the *whole* string onto each alias base path, so `@/src/auth/Auth.model` became `<pkg>/@/src/auth/Auth.model`, which does not exist. Resolution failed, `expected` came back empty, and an existing guard returned early. Alias imports were never flagged because they were never resolved.

`2.1.0` added proper wildcard handling, so alias imports resolve now. Once they do, this gate decides whether the rule reports:

```js
settings.maxDepth < (current.match(/\.\.[\\/]/g) || []).length
```

A specifier with no `../` counts `0`, so a negative `maxDepth` makes it true for every import in the project — including ones that are already absolute. `@/src/auth/Auth.model` then resolves back to itself, which is the identical-paths message you saw, and the fixer rewrites the specifier to itself, which is the circular fix.

While reproducing it I hit a second case of the same family that you did not mention — worth checking whether it also affects you. With `baseUrl` set, an import like `src/auth/Auth.model` gets reported as `Relative import path 'src/auth/Auth.model' should be replaced with '@/src/auth/Auth.model'`. It is not a relative import, and unlike the first case the paths differ, so `--fix` silently rewrites it.

The fix restricts the rule to actually-relative specifiers and guards the identity case. Verified end to end: both errors gone, `--fix` settles, and a genuine `../../../../src/auth/Auth.model` still reports and still gets rewritten to the alias form.

Thanks for the clear report — the version comparison is what made the cause findable.

**Comment by qdanik** [comment]
@westarne fix already delivered in v2.1.1
